bigquery/google/cloud/bigquery/table.py

```diff
@@ -473,8 +473,9 @@ def _build_resource(self):
             resource['schema'] = {
                 'fields': _build_schema_resource(self._schema)
             }
-        else:
-            raise ValueError("Set either 'view_query' or 'schema'.")
+        elif self.partitioning_type is None:
+            raise ValueError(
+                "Set either 'view_query' or 'schema' or 'partitioning_type'.")
```

Comment:
The autodetect parameter in table insert is useful for external tables (Google Drive, Sheets, Bigtable) where the data is stored outside of BigQuery.

I don't actually know what will happen when we try to create an autodetect external table with this client library. Looks to me like this exception will get thrown.

I understand the desire for easy unit testing, but this client library is not changing in exact sync with the backend API, so I'd prefer if this validation was done server-side.